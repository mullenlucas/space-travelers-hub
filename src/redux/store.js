import { combineReducers, configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rockets from './rocketsSlice';
import missions from './missionsSlice';

const reducer = combineReducers({
  rockets,
  missions,
});

const store = configureStore({
  reducer,
  middleware: [thunk, logger],
});

export default store;
