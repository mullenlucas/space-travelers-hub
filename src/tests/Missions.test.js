import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MissionPage from '../components/missions/missionsPage';

it('Mission page rendering', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MissionPage />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
