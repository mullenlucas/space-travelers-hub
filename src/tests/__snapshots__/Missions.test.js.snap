// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mission-container container"
>
  <div
    className="mission-table table-responsive"
  >
    <table
      className="table table-striped table-bordered"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Mission
          </th>
          <th
            scope="col"
          >
            Description
          </th>
          <th
            scope="col"
          >
            Status
          </th>
          <th
            scope="col"
          >
            Join
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
